### ISSUE TITLE
Throttle `reload-all` to avoid being blocked when reloading many feeds

### ISSUE THREAD

**Issue by 0xACE** [newsboat/newsboat#1800]
Newsboat 2.25.0

I have roughly 420 feeds that belong to google services. Whenever i hit `R` (`reload-all`) or `reload-time` triggers, it seems like i get blocked (`403`) because newsboat is accessing a lot of feeds over a short time period.

Currently I'm experimenting with `reload-time` to 6 hours just to see if it helps with not getting blocked by google...

I guess I'm looking for a way to reload a group of feeds or do them in batches  rather then all in one shot.

**Comment by Minoru** [comment]
Hmm. We are batching feeds by domain name:

https://github.com/newsboat/newsboat/blob/c67526dc90374b6b070b7ad02779db9c6a686ff0/src/reloader.cpp#L223-L240

I wasn't around when this code was written, but I think the goal is connection reuse. If we insert pauses in between each request, that should do the trick.

See also #976; it makes sense to move to a new runtime *before* introducing this change, but if that's too much, then simply fixing *this* issue is good too.

**Comment by 0xACE** [comment]
Yeah, in my case in particular: google aren't too happy about my behaviour. The ip ban lasts 12-24 hours. I'm not even sure if google would be cool with a slower churn rate, it feels like the problem was too many queries in a short period. Alas I will try a simple `sleep(1);`
in
https://github.com/newsboat/newsboat/blob/c67526dc90374b6b070b7ad02779db9c6a686ff0/src/reloader.cpp#L244-L249

Other than that my choices are:

1. Reduce my feeds list
2. spread newsboat over multiple machines and let each of them get chunks of the feeds (google was happy before the 400 mark)
3. Patch `newsboat` to not perform more than 100~ish fetches on `reload-all` or `reload-time`

**Comment by Auster-1** [comment]
Any news on that? I also got a rather long list of feeds which, upon running reload-all, gets some domains to time out.
